trade-finance-manager-ui/server/controllers/utilisation-reports/add-to-an-existing-payment/get-back-link-href.ts

```diff
@@ -0,0 +1,29 @@
+// "Recommended Security and Performance Max: 2048 CHARACTERS" - https://stackoverflow.com/a/48230425
+const MAX_URL_CHARACTERS = 2048;
+const ESTIMATED_BASE_URL_CHARACTERS = 100;
+
+export const getBackLinkUrl = (reportId: string, feeRecordIds: number[]): string => {
+  const urlWithoutParams = `/utilisation-reports/${reportId}`;
+
+  if (feeRecordIds.length === 0) {
+    return urlWithoutParams;
+  }
+
+  const commaSeparatedFeeRecordIds = feeRecordIds.join(',');
+
+  const feeRecordIdsURLParams = new URLSearchParams({
+    selectedFeeRecordIds: commaSeparatedFeeRecordIds,
+  }).toString();
+
+  const urlWithParams = `${urlWithoutParams}?${feeRecordIdsURLParams}`;
+
+  const urlWithParamsMaxLength = MAX_URL_CHARACTERS - ESTIMATED_BASE_URL_CHARACTERS;
+  if (urlWithParams.length > urlWithParamsMaxLength) {
```

Comment:
Fallback to sending the URL without params if the query params make the URL too long/unsafe. If the URL is too long the browser may truncate it, leading to a subset of the selected fee records being selected when we go back - want to avoid this. The query method should handle up to 278 seven digit fee record IDs before it reaches our max URL char limit.

Any thoughts on this additional check?

(2048-100 = 1948 chars, 1948 / 7 = 278 seven digit IDs)